### DISCUSSION TITLE
Fermata Auto app not showing up in Oneplus Nord CE 2 with Android 12

### DISCUSSION THREAD

**Discussion by srikanth2042** [AndreyPavlenko/Fermata#207] [Announcements]
Fermata auto app is not showing up in car’s screen once connected. I see fermata control app but there are no options to view YouTube.
Android version 12
Android auto version 8.0.623354-release
Developer mode is turned on and Unknown sources are allowed too in Android auto settings

**Comment by fcaronte** [comment]
yes is a know thing, unfortunately all custom android auto app won't work on any Oppo/Realme/OnePlus smartphone and we can't understand why. probably some firmware part give a bug and make it not to work with the usually trick

**Comment by srikanth2042** [comment]
It might be because oneplus phones run on oxygen os which is a layer on top of android os?

**Comment by Hirosashii** [comment]
> It might be because oneplus phones run on oxygen os which is a layer on top of android os?

If it helps at all, I have a OnePlus 7 Pro, and Fermata Auto _did_ work on it while I was OxygenOS version OP7Pro_02_BETA_5. It was also Android Auto 8.2.623913-release. I have since updated to Android 12 and now I have the same situation as srikanth2042. But I just wanted to make it known that as a OnePlus user, I can confirm that it did indeed work before under certain circumstances.

**Comment by Hirosashii** [comment]
I actually just downgraded to 11 to see if it would work again. It was a pain to try to get OP7Pro_02_BETA_5 working, so I ended up with **Oxygen OS 11.0.5.1.GM21AA** and it looks like Fermata Auto still works even here. It's just when I updated to 12 that it had an issue.

https://i.postimg.cc/D0zYcnPJ/Untitled.png
https://i.postimg.cc/QMr4ypqr/Screenshot-20221102-164324.jpg
https://i.postimg.cc/XqXQCxn9/IMG-20221102-165245.jpg

**Comment by srikanth2042** [comment]
@Hirosashii yes my oneplus phone is also running on Android 12, fermata auto is not working either. So the issue is with android 12 only.

**Comment by fcaronte** [comment]
> @Hirosashii yes my oneplus phone is also running on Android 12, fermata auto is not working either. So the issue is with android 12 only.

No fermata work on Android 12 and 13 too, the problem are Oppo based firmware like realme, OnePlus and Oppo phone we don't know why they aren't work

**Comment by AndreyPavlenko** [comment]
Try this - https://github.com/AndreyPavlenko/Fermata/discussions/122#discussioncomment-4307775

**Reply by AOH800** [reply]
Hi Andrey! The fix you mentioned only work for rooted OnePlus phones.
Is any chance to overcame this issue for those who don't want to root their phones.
I really appreciate your support. Thank you.

**Reply by AndreyPavlenko** [reply]
Unfortunately, I don't have any solution for unrooted devices. Otherwise, it would have already been implemented.

**Comment by desaeverg** [comment]
hello everyone i have a rooted oneplus 9 pro with android oxygenos 13.0. fermata control will be visible in android auto but here the function for youtube and web browser will not be visible. what am I overlooking? android car version 8.5.625224

anyone here had similar problems?

thanks in advance!
![IMG20230119195548](https://user-images.githubusercontent.com/91951880/213535795-d3aad22f-751e-4710-bdc4-2e0bf981ebe1.jpg)

**Reply by AndreyPavlenko** [reply]
https://github.com/AndreyPavlenko/Fermata/discussions/207#discussioncomment-4307790

**Comment by BHT456** [comment]
One plus 10R with Android 13  is having same issue. Only fermata control is showing in car screen . No fermata auto. No car stream YT.

**Comment by AOH800** [comment]
Thank for your response and for your work !

**Comment by fcaronte** [comment]
On Italian android auto group we discovered this trick and seem working for some Oppo, Relame and OnePlus user

Practically need to install a second package manager to force kinginstaller ask you which app want to use to install and then choose stock one that need to be the Google package installer with this package name
"com.google.android.packageinstaller" actually if stock have another name seem not to work.
To unlock the menu they install the Oppo package installer but never use it and choose for the stock one
Install this one
https://www.apkmirror.com/apk/oppo/package-installer-4/package-installer-4-3-0-2369-release/package-installer-3-0-2369-android-apk-download/download/?key=ee50e546529a21462096343668af022e100e969d
And then using kinginstaller choose the with with the Download arrow icon
![Screenshot_20230226-122123.png](https://user-images.githubusercontent.com/26366403/221407645-1ec9d9ff-1e05-4d43-89fb-4718a1300bb3.png)

**Reply by kathiravan619** [reply]
Thankyou sir, if anyone done in this method Please post a tutorial video, i tried but still not working ,

**Reply by AOH800** [reply]
> On Italian android auto group we discovered this trick and seem working for some Oppo, Relame and OnePlus user
> 
> Practically need to install a second package manager to force kinginstaller ask you which app want to use to install and then choose stock one that need to be the Google package installer with this package name "com.google.android.packageinstaller" actually if stock have another name seem not to work. To unlock the menu they install the Oppo package installer but never use it and choose for the stock one Install this one https://www.apkmirror.com/apk/oppo/package-installer-4/package-installer-4-3-0-2369-release/package-installer-3-0-2369-android-apk-download/download/?key=ee50e546529a21462096343668af022e100e969d And then using kinginstaller choose the with with the Download arrow icon ![Screenshot_20230226-122123.png](https://user-images.githubusercontent.com/26366403/221407645-1ec9d9ff-1e05-4d43-89fb-4718a1300bb3.png)

Yeeah !Finally !  I folowed the steps above and i have Fermata working again on my car screen using as phone OnePlus 9 Pro connected wireless at my car Kodiaq 2021 -MIB 3. 
I need to mention that : when i tried to install the package installer wich i dowloaded from the above link i received the message : it
 couldnt install because of getting in conflict with an allready installed package.

**Reply by AOH800** [reply]
![WhatsApp Image 2023-02-27 at 14 11 08 (1)](https://user-images.githubusercontent.com/85876500/221561875-0bdb1732-f12c-443e-b3b5-d927c5193fff.jpeg)
![WhatsApp Image 2023-02-27 at 14 11 08 (2)](https://user-images.githubusercontent.com/85876500/221561931-c760b0f0-6541-45e4-8afe-2d31aad9fbe6.jpeg)
![WhatsApp Image 2023-02-27 at 14 11 08](https://user-images.githubusercontent.com/85876500/221561957-896280d3-4611-4a1f-8a37-02d00baa9bc4.jpeg)

**Reply by fcaronte** [reply]
> ![WhatsApp Image 2023-02-27 at 14 11 08 (1)](https://user-images.githubusercontent.com/85876500/221561875-0bdb1732-f12c-443e-b3b5-d927c5193fff.jpeg)
> ![WhatsApp Image 2023-02-27 at 14 11 08 (2)](https://user-images.githubusercontent.com/85876500/221561931-c760b0f0-6541-45e4-8afe-2d31aad9fbe6.jpeg)
> ![WhatsApp Image 2023-02-27 at 14 11 08](https://user-images.githubusercontent.com/85876500/221561957-896280d3-4611-4a1f-8a37-02d00baa9bc4.jpeg)
> 

And then how to fix? Because on the grup I read talking about need to force close the "package installer" before to try install the other one. But I discover you can use any app that will show as apk installer, and for example you can use the new version of AAstore 2.1 and trick will work same

**Reply by AOH800** [reply]
What i did , i practically disregard the message and went ahead with King installer ... So i didnt fix anything !

**Reply by fcaronte** [reply]
> What i did , i practically disregard the message and went ahead with King installer ... So i didnt fix anything !

You need to click to the installer with the arrow icon

**Reply by AOH800** [reply]
Sorry to non mention ! This is what i did and it worked !

**Reply by rraghava10** [reply]
Did same mentioned as above and it is working..Thanks for the info. It really helped.. thank you so much...😀

**Comment by fcaronte** [comment]
I publish a mod of KingInstaller to support Oppo, Realme, and OnePlus trick without need to use another package installer. get here

https://github.com/fcaronte/KingInstaller/releases/tag/1.2

**Reply by BHT456** [reply]
What about those who have AAAD ?

**Reply by schumi2004** [reply]
Its not working for me. OnePlus Nord Pixel Experience 13

**Comment by BHT456** [comment]
What about those who have AAAD not kinginstaller

**Reply by fcaronte** [reply]
I'll write the AAAD to implement this trick on his app to, but for now you can install my version of KingInstaller, then open it and enable he oppo flag, and you can go to aaad and install what's you want and will appear a popup with kinginstaller say you to touch the other package installer, practically show this popup is the trick on Oppo Relame OnePlus problem, so remember to NOT save as default your choice, then if you don't need to see always the fake kinginstaller icon when you install any apk not related to android auto, you can remove the flag in the settings and should go back to install directly as Classic way

**Reply by Kymatozni4ek** [reply]
Подскажите пожалуйста что я не так делаю 

Uploading 2_5204276322842455297.mp4…

**Reply by BHT456** [reply]
Thanks Fcarnote !!!  It works for me
